pkg/cloud/services/networking/securitygroups.go

```diff
@@ -167,9 +175,65 @@ func (s *Service) generateDesiredSecGroups(openStackCluster *infrav1.OpenStackCl
 		Rules: workerRules,
 	}
 
+	allNodesRules, err := getAllNodesRules(remoteManagedGroups, openStackCluster.Spec.ManagedSecurityGroups.AllNodesSecurityGroupRules)
+	if err != nil {
+		return desiredSecGroups, err
+	}
+	desiredSecGroups[allNodesSuffix] = infrav1.SecurityGroup{
+		Name:  secGroupNames[allNodesSuffix],
+		Rules: allNodesRules,
+	}
+
 	return desiredSecGroups, nil
 }
 
+// getAllNodesRules returns the rules for the allNodes security group.
+func getAllNodesRules(remoteManagedGroups []string, allNodesSecurityGroupRules []infrav1.SecurityGroupRule) ([]infrav1.SecurityGroupRule, error) {
+	rules := make([]infrav1.SecurityGroupRule, 0, len(allNodesSecurityGroupRules))
+	for _, rule := range allNodesSecurityGroupRules {
+		if err := validateAllNodesRule(rule); err != nil {
+			return nil, err
+		}
+		for _, group := range remoteManagedGroups {
+			r := rule
+			r.RemoteGroupID = group
+			rules = append(rules, r)
+		}
+	}
+	return rules, nil
+}
+
+// validateAllNodesRule validates that the rule is valid for the allNodes security group.
+func validateAllNodesRule(rule infrav1.SecurityGroupRule) error {
+	if rule.RemoteGroupID != "" {
+		return fmt.Errorf("allNodesRule: remoteGroupID must not be set")
+	}
+	if rule.RemoteIPPrefix != "" {
+		return fmt.Errorf("allNodesRule: remoteIPPrefix must not be set")
+	}
+	if rule.RemoteManagedGroups != nil {
+		for _, group := range rule.RemoteManagedGroups {
+			switch group {
+			case controlPlaneSuffix, workerSuffix, bastionSuffix:
+				continue
+			default:
+				return fmt.Errorf("allNodesRule: remoteManagedGroups must be either \"controlplane\", \"worker\" or \"bastion\"")
+			}
+		}
+	}
+
+	// default to the managed groups if no remoteManagedGroups are set
+	// Here we want to figure out what to do with defaults, several options:
+	// - pass a map of managed groups with their suffix and IDs
+	// - use a keyword like "all" if no remoteManagedGroups are set and later in getAllNodesRules replace it with the managed groups IDs passed in the first arg
+	// - don't provide a default and force the user to set remoteManagedGroups
+	if len(rule.RemoteManagedGroups) == 0 {
+		rule.RemoteManagedGroups = []string{controlPlaneSuffix, workerSuffix, bastionSuffix}
```

Comment:
this block will be removed, it doesn't belong here but it'll be done in `getAllNodesRules`.